app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalArticleController.php

```diff
@@ -47,7 +51,13 @@ class KnowledgeManagementPortalArticleController extends Controller
 {
     public function show(KnowledgeBaseCategory $category, KnowledgeBaseItem $article): JsonResponse
     {
+        if (! Session::exists('guest_id') && ! Auth::guard('contact')->check() && ArticleWasHelpful::active()) {
+            $portalGuest = PortalGuest::create();
+            Session::put('guest_id', $portalGuest->id);
+        }
         $article->increment('portal_view_count');
+        $userType = Session::exists('guest_id') ? 'PortalGuest' : 'Contact';
```

Comment:
This type string seems strange. I would expect something like `portal-guest` and `contact` to match the MorphMap.

In fact, you should probably get the Morph Map value from the classes themselves rather than static strings.

Reply:
Done